src/main/java/javax/jmdns/impl/JmDNSImpl.java

```diff
@@ -1018,12 +1018,15 @@ public void registerService(ServiceInfo infoAbstract) throws IOException {
         info.addAddress(_localHost.getInet4Address());
         info.addAddress(_localHost.getInet6Address());
 
+        this.waitForAnnounced(DNSConstants.SERVICE_INFO_TIMEOUT);
+
         this.makeServiceNameUnique(info);
         while (_services.putIfAbsent(info.getKey(), info) != null) {
             this.makeServiceNameUnique(info);
         }
 
         this.startProber();
+        info.waitForAnnounced(DNSConstants.SERVICE_INFO_TIMEOUT);
```

Comment:
Why a second time here?

Reply:
Sorry, I will have to take a look to verify why is this shown as it is part of my commit. It is not. I presume it was there before my changes and I just moved it.